src/main/java/eu/tsystems/mms/tic/testerra/plugins/xray/synchronize/AbstractXrayResultsSynchronizer.java

```diff
@@ -352,16 +359,62 @@ public void onTestStatusUpdate(TestStatusUpdateEvent event) {
          */
         currentTestIssues.stream()
                 .peek(issue -> xrayMapper.updateTest(issue, methodContext))
-                .map(XrayTestExecutionImport.TestRun::new)
-                .peek(test -> updateTestImport(test, methodContext))
+                .map(issue -> {
+                    XrayTestExecutionImport.TestRun run = new XrayTestExecutionImport.TestRun(issue.getKey());
+                    this.updateTestInfoForImport(run, issue, methodContext);
```

Comment:
This could have been another peek, for laconeia.

Reply:
Hm... the content of `map()` is to complex.

Reply:
Should work by constructing `new XrayTestExecutionImport.TestRun(issue.getKey()` inside `updateTestInfoForImport`, thereby eliminating its first parameter. The method is private.

```java
[...]

	// Update test by mapper, convert to import entity and add to sync queue
	currentTestIssues.stream()
			.peek(issue -> xrayMapper.updateTest(issue, methodContext))
			.map(issue -> this.updateTestInfoForImport(issue, methodContext))
			.peek(testRun -> updateTestRunForImport(testRun, methodContext))
			.forEach(testRunSyncQueue::add);

	if (testRunSyncQueue.size() >= xrayConfig.getSyncFrequencyTests()) {
		flushSyncQueue();
	}
}

/**
 * Update the Info object for creating or updating Xray tests.
 * If no Info object is defined, the Xray test will not be updated.
 */
private XrayTestExecutionImport.TestRun updateTestInfoForImport(XrayTestIssue issue, MethodContext methodContext) {
	if (this.getXrayMapper().shouldCreateNewTest(methodContext)) {
		XrayTestExecutionImport.TestRun.Info info = new XrayTestExecutionImport.TestRun.Info();
		info.setDescription(issue.getDescription());
		info.setSummary(issue.getSummary());
		info.setLabels(issue.getLabels());
		info.setDefinition(issue.getSummary());
		info.setType(TestType.AutomatedGeneric);
		info.setProjectKey(issue.getProject().getKey());
		
		// The test's test type needs to be {@link TestType.Manual} to support test steps.
		info.setType(TestType.Manual);

		List<TestStep> testerraTestSteps = methodContext.readTestSteps().collect(Collectors.toList());
		for (TestStep testerraTestStep : testerraTestSteps) {
			if (testerraTestStep.isInternalTestStep()) {
				continue;
			}
			// The test steps definitions
			final XrayTestExecutionImport.TestStep importTestStep = new XrayTestExecutionImport.TestStep();
			importTestStep.setAction(testerraTestStep.getName());
			// We always expect the step to pass
			importTestStep.setResult(XrayTestExecutionImport.TestRun.Status.PASS.toString());
			info.addStep(importTestStep);
		}

		final var testRun = new XrayTestExecutionImport.TestRun(issue.getKey());
		testRun.setTestInfo(info);
		return testRun;
	}
}
```

... FWIW: This is not really function-al code to begin with. It might instead be better to not-Stream this and for-each it instead, to avoid the red herring:
```java
// Update test by mapper, convert to import entity and add to sync queue
for (final var issue : currentTestIssues) {
	xrayMapper.updateTest(issue, methodContext);
	final var run = new XrayTestExecutionImport.TestRun(issue.getKey());
	this.updateTestInfoForImport(run, issue, methodContext);
	this.updateTestRunForImport(run, methodContext);
	testRunSyncQueue.add(run);
}
```
(without changes in the `update[...]` functions)

... But feel free to reject any of these ideas - it's not particularly important and I can also live with this part as-is.

Reply:
Good point. I moved the creation of a new `TestRun` into the method.